### DISCUSSION TITLE
How task is being dequed?

### DISCUSSION THREAD

**Discussion by parthAjagiya** [hibiken/asynq#792] [Q&A]
What decides "its time to deque the task"? Is it Redis (ttl kinda logic)?

My usecase context:
We are using asynq to schedule a notification. Our service is hosted on AWS across 10 containers and uses common Redis. On spin off service, we schedule a task with same taskId across the all containers, at scheduled time handler gets executed and that handler again calls code to schedule next task. As taskId is same for all containers, only one task gets scheduled and we only sends notification single time.
But we recently had issue where notification was sent multiple times. Upon checking it was sent from 7 different containers on different times than scheduled time.
I checked that task was scheduled correctly.
So now I am suspecting that there should be some culprit at deque side. So if asynq relies upon Redis to decide time to deque then there should be issue with Redis.

**Comment by kamikazechaser** [comment]
See `processor.exec` for the implementation.
